Allow kwargs to pass to lldl

JeffFessler opened the pull request:
This should address #30.
I have not yet added and tests to confirm the additional functionality.
@learning-chip does this seem like the appropriate extension and can you suggest a test?

mohdibntarek commented:
The implementation looks correct to me. Please add a test and let's this get this merged.

JeffFessler commented:
I added a minimal test.
I also made it more clear to future developers which functions come from which packages.
There was also an unused `where T` that julia v1.8 complained about that I removed.
And I bumped the version to 0.6.

JeffFessler commented:
@mohamed82008 this is ready for you to review

mohdibntarek commented on src/incompletecholesky.jl:
```diff
@@ -2,26 +2,36 @@ function assert_pd(d, α)
 
+"""
+    CholeskyPreconditioner(A, memory=2; kwargs...)
+Construct
```
We should use JuliaFormatter at some point to have consistent formatting. The lack of an empty line before `Construct` and short docstring lines strike me as odd but it's not a blocker. I am happy to follow any standard automated format.
